backend/src/v5/middleware/dataConverter/inputs/users.js

```diff
@@ -158,15 +161,28 @@ Users.validateForgotPasswordData = async (req, res, next) => {
 
 		try {
 			const usernameOrEmail = req.body.user;
-			const { user, customData: { sso } } = await getUserByUsernameOrEmail(usernameOrEmail, { user: 1, _id: 0, 'customData.sso': 1 });
+			const { user, customData: { sso, email, firstName } } = await getUserByUsernameOrEmail(usernameOrEmail, {
+				user: 1,
+				_id: 0,
+				'customData.sso': 1,
+				'customData.firstName': 1,
+				'customData.email': 1,
+			});
 
 			if (sso) {
+				await sendEmail(mailTemplates.FORGOT_PASSWORD_SSO.name, email,
+					{
+						email,
+						username: user,
+						firstName,
```

Comment:
do we need to send the firstName and email ?

Reply:
firstName is required as part of the template so we can write the "hello %firstname%" part. https://github.com/3drepo/3drepo.io/blob/master/backend/src/v5/services/mailer/templates/baseTemplate.js#L27

Email.. i guess not.